### PR TITLE
GitHub Issue NOAA-EMC/GSI#259. Add assimilated flag to observations in osense file

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Changes to the following files:
src/enkf/enkf_obsmod.f90
src/enkf/enkf_obs_sensitivity.f90
src/enkf/enkf.f90

In serial filter, creates array size of number of observations and sets whether they are assimilated or not, and then sends to osense file if ran in efsoi mode.

### CODE REVIEW

**Comment by MichaelLueken-NOAA** [comment]
@AndrewEichmann-NOAA - Please reduce the number of commit messages from 23 to 1 using the following:

1) git clone --recursive git@github.com:AndrewEichmann-NOAA/GSI.git update
2) cd update
3) git reset --hard HEAD~1
4) git reset --hard HEAD~1
5) git reset --hard HEAD~1
6) git reset --hard HEAD~1
7) git reset --hard HEAD~1
8) git reset --hard HEAD~1
9) git reset --hard HEAD~1
10) git reset --hard HEAD~1
11) git reset --hard HEAD~1
12) git reset --hard HEAD~1
13) git reset --hard HEAD~1
14) git reset --hard HEAD~1
15) git reset --hard HEAD~1
16) git reset --hard HEAD~1
17) git reset --hard HEAD~1
18) git reset --hard HEAD~1
19) git reset --hard HEAD~1
20) git reset --hard HEAD~1
21) git reset --hard HEAD~1
22) git reset --hard HEAD~1
23) git reset --hard HEAD~1
24) git remote add upstream git@github.com:NOAA-EMC/GSI.git
25) git remote update
26) git merge upstream/master
27) git checkout EXP-efso_fv3_PR_src_enkf_112021
28) git checkout master
29) git merge --squash EXP-efso_fv3_PR_src_enkf_112021
30) git commit -m `"GitHub Issue NOAA-EMC/GSI#259. Add assimilated flag to observations in osense file."`
31) git push origin master --force

Also, please let me know who you would like to have review the changes of this PR.

**Comment by AndrewEichmann-NOAA** [comment]
Actions described should be done.
Suitable reviewers would be Cathy and Rahul.

File: src/enkf/enkf.f90 [hunk 1]
```diff
@@ -776,6 +791,8 @@ subroutine enkf_update()
       if (covl_fact < 0.99) print *,'mean covl_fact = ',covl_fact
       if (nskip > 0) print *,nskip,' out of',nobstot,'obs skipped,',nuse,' used'
       if (nsame > 0) print *,nsame,' out of', nobstot-nskip,' same lat/long'
+      if (nuseconvoz > 0 ) print *,nuseconvoz,' out of',nobs_conv + nobs_oz ,'convobs used'
```
Comment (CatherineThomas-NOAA) [inline src/enkf/enkf.f90:794]: Why are the conventional and ozone obs lumped together here?  Why not keep them separate as is done elsewhere?

Comment (AndrewEichmann-NOAA) [reply src/enkf/enkf.f90:794]: They are separated that way in the osense file as the bias predictors are saved for the satellite observations, and the conventional and ozone obs are lumped together in the existing counts (as used in the osense header), so this is what I used for troubleshooting. There's no particular reason why the ozone and conventional observations couldn't be parted out here.

**Review by jswhit2 — APPROVED** [review]

File: src/enkf/enkf.f90 (hunk 1 above)
Comment (CatherineThomas-NOAA) [reply src/enkf/enkf.f90:794]: Okay that makes sense.  You can leave it.

**Review by CatherineThomas-NOAA — APPROVED** [review]

**Comment by MichaelLueken-NOAA** [comment]
Hi @AndrewEichmann-NOAA, while merging the authoritative master to your fork's master, you added an extra commit message.  Please use the following to bring the commit messages back to one and update your fork's master to the latest authoritative master:

1) git reset --hard HEAD~1
2) git remote add upstream https://github.com/NOAA-EMC/GSI.git
3) git remote update
4) git rebase upstream/master
5) git submodule update
6) git commit --amend (please save and close the commit message window without altering the commit message)
7) git push origin master --force

Please note, if you have already added the upstream in your local copy, then you can skip step 2).

If you have any questions, please let me know.

**Comment by AndrewEichmann-NOAA** [comment]
@MichaelLueken-NOAA The steps should be done.

**Comment by MichaelLueken-NOAA** [comment]
@AndrewEichmann-NOAA Everything looks good.  Once a small bug fix passes the review committee, I should be able to submit this PR early next week.

**Comment by MichaelLueken-NOAA** [comment]
The due date for the review committee has passed with no feedback, so I will now give final approval to these changes and merge them to the authoritative master.

**Pull request merged**